### PR TITLE
Make number of training events a telescope parameter

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Review by nbiederbeck — APPROVED** [review]

File: ctapipe/core/traits.py [hunk 1]
```diff
@@ -636,7 +643,7 @@ class IntTelescopeParameter(TelescopeParameter):
 
     def __init__(self, **kwargs):
         """Create a new IntTelescopeParameter"""
-        super().__init__(trait=Int(), **kwargs)
+        super().__init__(trait=Integer(), **kwargs)
```
Comment (LukasNickel) [inline ctapipe/core/traits.py:646]: Why do we have both `Int` and `Integer`?

Comment (maxnoe) [reply ctapipe/core/traits.py:646]: These are aliases in traitlets, exactly the same object....

File: ctapipe/tools/train_energy_regressor.py [hunk 2]
```diff
@@ -143,8 +145,8 @@ def _read_table(self, telescope_type):
             self.log.warning("Dropping non-predictable events.")
             table = table[valid]
 
-        if self.n_events is not None:
-            n_events = min(self.n_events, len(table))
+        if self.n_events.tel[telescope_type] is not None:
+            n_events = min(self.n_events.tel[telescope_type], len(table))
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply ctapipe/tools/train_energy_regressor.py:149]: I think that should even be an error.

**Comment by maxnoe** [comment]
@kosack @LukasNickel @nbiederbeck comments addressed

**Review by LukasNickel — APPROVED** [review]

**Pull request merged**